src/ng-select/items-list.ts

```diff
@@ -248,7 +248,10 @@ export class ItemsList {
 
     private _groupBy(items: NgOption[], prop: string): { [index: string]: NgOption[] } {
         const groups = items.reduce((grouped, item) => {
-            const key = item.value[prop];
+            const key = prop.split('.').reduce(
+                function (accumulator, current) {
+                    return accumulator[current];
+                }, item.value);
             grouped[key] = grouped[key] || [];
             grouped[key].push(item);
```

Comment:
It would be more flexible if groupbBy prop could by string or function and performance would be better :)

```ts
private _groupBy(items: any, prop: string | Function): { [index: string]: NgOption[] } {
   const isPropFn = prop instanceof Function;
   const groups = items.reduce((grouped, item) => {
      const key = isPropFn ? prop.apply(this, [item]) : item[prop];
      grouped[key] = grouped[key] || [];
      grouped[key].push(item);
      return grouped;
   }, {});
  return groups;
}
```

Also please add test case for custom groupBy with function. Thanks 👍